src/features/Faucet/Faucet.tsx

```diff
@@ -0,0 +1,283 @@
+import React, { useContext, useState } from 'react';
+
+import { Heading, Icon, Input, Tooltip } from '@mycrypto/ui';
+import { Field, FieldProps, Form, Formik, FormikProps } from 'formik';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import styled from 'styled-components';
+
+import questionToolTip from '@assets/images/icn-question.svg';
+import receiveIcon from '@assets/images/icn-receive.svg';
+import {
+  AccountSelector,
+  Button,
+  ExtendedContentPanel,
+  InlineMessage,
+  NewTabLink,
+  TxReceipt
+} from '@components';
+import { getKBHelpArticle, KB_HELP_ARTICLE, ROUTE_PATHS } from '@config';
+//import { StoreContext } from '@services/Store';
+import { StoreContext, useAssets, useContacts } from '@services/Store';
+import translate, { translateRaw } from '@translations';
+import { IAccount as IIAccount, InlineMessageType, StoreAccount } from '@types';
+import { noOp } from '@utils';
+
+import { Error } from './components';
+import {
+  makeTxConfig,
+  makeTxReceipt,
+  possibleSolution,
+  regenerateChallenge,
+  requestChallenge,
+  solveChallenge
+} from './helpers';
+
+// Legacy
+
+const SLabel = styled.label`
+  margin-bottom: 8px;
+  color: #333333;
+  font-weight: normal;
+`;
+
+const Fieldset = styled.fieldset`
+  margin-bottom: 15px;
+`;
+
+const CodeHeader = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const CodeHeading = styled(Heading)`
+  margin-top: 8px;
+`;
+
+const LearnMoreArrowWrapper = styled.span`
+  vertical-align: middle;
+  margin-left: 2px;
+`;
+
+const LearnMoreArrow = styled(Icon)`
+  transform: rotate(180deg);
+  width: 14px;
+  height: 14px;
+`;
+
+const NoTestnetAccounts = styled(InlineMessage)`
+  margin-bottom: 20px;
+  a {
+    color: var(--color-link-color);
+    text-decoration: none;
+  }
+  a:hover {
+    color: var(--color-link-color);
+  }
+`;
+
+const IncorrectResponse = styled(InlineMessage)`
+  margin-top: 10px;
+`;
+
+const Description = styled.p`
+  margin-bottom: 30px;
+`;
+
+const CenterText = styled.div`
+  margin-top: 30px;
+  text-align: center;
+`;
+
+const RequestButton = styled(Button)`
+  width: 100%;
+`;
+
+const SubmitCaptchaButton = styled(Button)`
+  width: 100%;
+  margin-top: 30px;
+`;
+
+const faucetNetworks = ['Ropsten', 'Rinkeby', 'Kovan', 'Goerli'];
+
+export function Faucet({ history }: RouteComponentProps<{}>) {
```

Comment:
Instead of exporting with `withRouter`, you can use `useHistory` inside the component